internal/cloudsql/instance.go

```diff
@@ -190,7 +202,7 @@ func (i *Instance) scheduleRefresh(d time.Duration) *refreshResult {
 	res := &refreshResult{}
 	res.ready = make(chan struct{})
 	res.timer = time.AfterFunc(d, func() {
-		ctx := context.Background() // TODO: store this in Instance
+		ctx := i.ctx
```

Comment:
Should we check if this context is canceled here to avoid scheduling another refresh operation otherwise?

Reply:
Likewise, should we be stopping the current and next refresh results as well?

Reply:
I added a check to prevent the new refresh from being scheduled. It didn't necessarily seem worth it to try to stop the AfterFunc from triggering when scheduled, since the context is closed it'll be a no-op.